src/Atmos/Model/hyperdiffusion.jl

```diff
@@ -58,25 +58,25 @@ function diffusive!(
 
 """
   NoHyperDiffusion <: HyperDiffusion
-Defines a default hyperdiffusion model with zero diffusive fluxes. 
+Defines a default hyperdiffusion model with zero diffusive fluxes.
 """
 struct NoHyperDiffusion <: HyperDiffusion end
 
 """
   StandardHyperDiffusion{FT} <: HyperDiffusion
 Horizontal hyperdiffusion methods for application in GCM and LES settings
-Timescales are prescribed by the user while the diffusion coefficient is 
+Timescales are prescribed by the user while the diffusion coefficient is
 computed as a function of the grid lengthscale.
 """
 struct StandardHyperDiffusion{FT} <: HyperDiffusion
     τ_timescale::FT
 end
 
-vars_aux(::StandardHyperDiffusion, FT) = @vars(Δ::FT)
-vars_gradient(::StandardHyperDiffusion, FT) =
+vars_state_auxiliary(::StandardHyperDiffusion, FT) = @vars(Δ::FT)
+vars_state_transform(::StandardHyperDiffusion, FT) =
     @vars(u::SVector{3, FT}, h_tot::FT)
 vars_gradient_laplacian(::StandardHyperDiffusion, FT) =
-    @vars(u::SVector{3, FT}, h_tot::FT)
+    @vars(Δu::SVector{3, FT}, Δh_tot::FT)
```

Comment:
Is this to compute gradients of the Laplacian?

Reply:
yeah, the input to the hyperdiffusive variables is `∇Δstate`. 

The notation here is a little confusing because this variable is both where the Laplacian is stored and where the gradient is stored after taking the Laplacian. Same thing with the old`vars_gradient`/`vars_transform`. It initially holds the thing you want to take the derivative of, filled in `gradvariables!`/`transform_pre_gradient!`, and then holds the derivative of it when being called in `diffusive!`/`transform_post_gradient!`. 
Because of this ambiguity/dual role, it is a bit hard to come up with a good name for these things, and it's only worse in the case of hyperdiffusion, as you have even more reuse of variables for storage purposes. 

For second-order/diffusive terms, I've decided to go with calling the thing before the gradient the transform state and the variables actually called in `flux_second_order!` the gradient state. Ideally, we will come up with some similar notation for hyperdiffusion... but that's a separate PR.

Reply:
This seems to be causing an issue @mwarusz. Do these need to have the exact same name as the `vars_state_gradient` in the main model?

Reply:
The only have to match for a given model, i.e. `vars_gradient_laplacian(m)` has to be a subset of `vars_state_gradient(m)`.

Reply:
and is this just for the hyperdiffusion model or does it also have to match the main atmos model?

Reply:
They way that hyperdiffusion is implemented is in the `Atmos` model the names should only need to match the hyperdiffusion model. However, this means that we are recomputing gradients if hyperdiffusion is used. Not sure if there was a specific reason to do that @akshaysridhar ?